src/module.rs

```diff
@@ -1,10 +1,22 @@
 use crate::support::int;
+use crate::support::MaybeBool;
 use crate::support::Opaque;
 use crate::Context;
 use crate::Local;
+use crate::String;
 use crate::Value;
 
-extern "C" {}
+// Ideally the return value would be Option<Local<Module>>... but not FFI-safe
+type ResolveCallback =
+  extern "C" fn(Local<Context>, Local<String>, Local<Module>) -> *mut Module;
```

Comment:
@piscisaureus Have you thought through how this works with respect to lifetimes?

Reply:
Yes. This looks good.